File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character, Integer> map = new HashMap<>();
        int temp;
        for(int i=0; i<magazine.length(); i++){
            if(map.containsKey(magazine.charAt(i))){
                temp = map.get(magazine.charAt(i));
                map.put(magazine.charAt(i), temp+1);
            }
            else{
            map.put(magazine.charAt(i), 1);
            }
        }
        for(int i=0; i<ransomNote.length(); i++){
            if(!map.containsKey(ransomNote.charAt(i)) || map.get(ransomNote.charAt(i)) == 0) {

                return false;
            }
            else{
                temp = map.get(ransomNote.charAt(i));
                map.put(ransomNote.charAt(i), temp-1);
            }
        }
        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        counter_magazine={}
        
        for i  in range(0,len(magazine)):
            temp=magazine[i]
            if temp not in counter_magazine:
                counter_magazine[temp]=1
            else:
                counter_magazine[temp]+=1
                    
        for i in range(0,len(ransomNote)):
            temp = ransomNote[i]
            if temp not in counter_magazine:
                return False
            else:
                counter_magazine[temp]-=1
                if counter_magazine[temp]<0:
                    return False
        return True
                    
                    
                    
                    
        
            